### DISCUSSION TITLE
maven.fabricmc.net is too slow and I cannot build a mod

### DISCUSSION THREAD

**Discussion by IAmREGE** [FabricMC/fabric-api#3725] [Mod Dev Support]
### Description

I am in **the PRC**, and with a poor Internet connectivity environment.  
As the title says, `./gradlew build` here is always with a `Connect timed out`.  
So I tried to find solutions, but none of these helped:
* [如何加速Fabric模组的构建 - FabricMCCN](https://fabricmc.cn/2021/06/28/%E5%A6%82%E4%BD%95%E5%8A%A0%E9%80%9FFabric%E6%A8%A1%E7%BB%84%E7%9A%84%E6%9E%84%E5%BB%BA/)
* <https://www.mcbbs.net/forum.php?mod=viewthread&tid=1239261>\(Referenced by [this page](https://fabricmc.net/wiki/zh_cn:tutorial:setup), but now it is obsolete\)
* Modifying `repositories` in `build.gradle`(the actual behaviour of Loom via `./gradlew build` is just to ignore them)

To ping maven.fabricmc.net 10000 times could receive a packet loss ratio with approximately 71%.
To download an 1.3MiB yarn jar in maven.fabricmc.net took about 4200 second and needed 8 reconnection.
# System Environment
### IDE: I think this is not important as I run `./gradlew build` in bash.

# Some Questions
* I heard that the downloads including intermediary uses a hard-coded `maven.fabricmc.net`. Does Fabric Loom support mirrors instead of hard-coded `maven.fabricmc.net`?
* Does Fabric Loom support partial content transfer?
* Is there any way to let Fabric Loom recognize manually-downloaded files (like `intermediary-1.20.6-v2.jar`)?

**Comment by davidromrell** [comment]
not stating this is correct, try doing network reset.
secondly try using IntelliJ it is the recommended supported tool,  KWrite is unsupported.

**Comment by Crystal0404** [comment]
你需要配置镜像下载源, [此页面](https://fabricmc.net/wiki/zh_cn:tutorial:setup)中的内容非常有效

同时建议配置上[阿里云的maven](https://developer.aliyun.com/mvn/guide)镜像, 也可以提升你的下载速度

**If you are not a developer from China, do not use them**

**Reply by davidromrell** [reply]
thank you for the info.

**Reply by IAmREGE** [reply]
I have already mentioned <https://fabricmc.net/wiki/zh_cn:tutorial:setup> which is useless for me.

**Reply by davidromrell** [reply]
it doesn't matter if the documentation does not work that is the proper way to do it almost all other ways will not work no matter how much effort.

**Reply by davidromrell** [reply]
@IAmREGE 
here are the only alternatives:
1. https://eclipse.dev/che/features/  
2. https://theia-ide.org/
3. https://eclipse.dev/jetty/ 
4. https://projects.eclipse.org/proposals/jakarta-commons

**Reply by Crystal0404** [reply]
It seems that using VPN is the best solution

**Reply by IAmREGE** [reply]
问题已经更新，请再阅读一遍。
